src/controllers/ScrollRestorationController.ts

```diff
@@ -0,0 +1,82 @@
+import { Assert } from '../misc/Assert';
+import { ResultListEvents, IDisplayedNewResultsEventArgs } from '../events/ResultListEvents';
+import { QueryStateModel } from '../models/QueryStateModel';
+import { $$ } from '../utils/Dom';
+import { Logger } from '../misc/Logger';
+
+/**
+ * This class is instantiated on the root of {@link SearchInterface}
+ * if the {@link SearchInterface.options.enableScrollRestoration} option is set to true.
+ *
+ * Its job is to wait until the search results are displayed and the page height is long enough to
+ * restore to the last scroll position. It'll wait for up to `restorationTimeOutInMs` milliseconds for the page to reach its height.
+ *
+ * This is useful when back and forth navigation between the search interface and results is needed on the same tab.
+ *
+ */
```

Comment:
The JSDoc like this will make the module/class documented "officially" on our reference documentation website, automatically.

Since, as you explained in your PR, this is essentially "internal" logic that clients won't have to configure at all ideally, then we should not be documenting it like this.

The main reason as for why we should not document it like this is because, once it appears "officially" on the reference website, it means that we have to take great care of never breaking backward compatibility. It also means that we think clients or implementer would have to directly interact with the module/class. This is not the case here, since it's an internal module.

And so, since we want to keep the possibility of changing the implementation however we see fit, as an internal module, we should not add such JSDoc.